### ISSUE TITLE
Labels persist over a polygon layer at full opacity

### ISSUE THREAD

**Issue by tristen** [mapbox/mapbox-gl-js#2909]
See http://bl.ocks.org/tristen/1f708402d46823051126a65c1ee30776 which reproduces the issue. I have a [GeoJSON object](https://gist.github.com/anonymous/b2c69192450ceb7e6e87b7c10db45fda) that encompasses the entire earth but if the layer style is at full opacity, not all layers are covered.

**Comment by tristen** [comment]
Interestingly, if you set a value other than 1 for `fill-opacity` labels are correctly under the layer.

**Comment by peterqliu** [comment]
possibly related https://github.com/mapbox/mapbox-gl-js/issues/2846

**Comment by andrewharvey** [comment]
Possibly this could be this issue https://github.com/mapbox/mapbox-gl-js/issues/2074 ?

> Interestingly, if you set a value other than 1 for fill-opacity labels are correctly under the layer.

Thank you for noticing this! I've been struggling with this issue for a while, but only after reading your comment now I realise I can just use opacity 0.99 as a workaround.

**Comment by mollymerp** [comment]
Oof. I'll take a look and see if this bug was introduced recently or if it likely related to the depth-testingissue #2074

**Comment by lucaswoj** [comment]
This looks like a duplicate of https://github.com/mapbox/mapbox-gl-js/issues/2074
